cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -1072,6 +1122,41 @@ private ProgramId validateProgramForRunRecord(String namespaceName, String appNa
   }
 
   /**
+   * Helper method to update log levels for Worker, Flow, or Service.
+   */
+  private void updateLogLevels(ProgramId programId, Map<String, LogEntry.Level> logLevels,
+                               @Nullable String component, @Nullable String runId) throws Exception {
+    ProgramRuntimeService.RuntimeInfo runtimeInfo = findRuntimeInfo(programId, runId).iterator().next();
+    if (runtimeInfo != null) {
+      LogLevelUpdater logLevelUpdater = getLogLevelUpdater(runtimeInfo);
+      logLevelUpdater.updateLogLevels(logLevels, component);
+    }
+  }
+
+  /**
+   * Helper method to reset log levels for Worker, Flow or Service.
+   */
+  private void resetLogLevels(ProgramId programId, @Nullable Set<String> loggerNames,
+                              @Nullable String component, @Nullable String runId) throws Exception {
+    ProgramRuntimeService.RuntimeInfo runtimeInfo = findRuntimeInfo(programId, runId).iterator().next();
+    if (runtimeInfo != null) {
+      LogLevelUpdater logLevelUpdater = getLogLevelUpdater(runtimeInfo);
+      logLevelUpdater.resetLogLevels(loggerNames == null ? Collections.<String>emptySet() : loggerNames, component);
```

Comment:
Why allows passing in `null` as `loggerNames`? I don't think caller will every pass in null.